csi/volumeutils.py

```diff
@@ -940,82 +941,11 @@ def generate_client_volfile(volname):
 
     VOL_DATA[volname]["hash"] = current_hash
 
-    # Tricky to get this right, but this solves all the elements of distribute in code :-)
-    data['dht_subvol'] = []
-    decommissioned = []
-    if data["type"] == "Replica1":
-        for brick in data["bricks"]:
-            brick_name = "%s-client-%d" % (data["volname"], brick["brick_index"])
-            data["dht_subvol"].append(brick_name)
-            if brick.get("decommissioned", "") != "":
-                decommissioned.append(brick_name)
-    else:
-        count = 3
-        if data["type"] == "Replica2":
-            count = 2
-
-        if data["type"] == "Disperse":
-            count = data["disperse"]["data"] + data["disperse"]["redundancy"]
-            data["disperse_redundancy"] = data["disperse"]["redundancy"]
-
-        data["subvol_bricks_count"] = count
-        for i in range(0, int(len(data.get("bricks", [])) / count)):
-            brick_name = "%s-%s-%d" % (
-                data["volname"],
-                "disperse" if data["type"] == "Disperse" else "replica",
-                i
-            )
-            data["dht_subvol"].append(brick_name)
-            if data.get("bricks", [])[(i * count)].get("decommissioned", "") != "":
-                decommissioned.append(brick_name)
-
-    data["decommissioned"] = ",".join(decommissioned)
-    template_file_path = os.path.join(
-        TEMPLATES_DIR,
-        "%s.client.vol.j2" % data["type"]
-    )
-    client_volfile = os.path.join(
-        VOLFILES_DIR,
-        "%s.client.vol" % volname
-    )
-    content = ""
-    with open(template_file_path) as template_file:
-        content = template_file.read()
-
-    Template(content).stream(**data).dump(client_volfile)
     return True
 
 
-def send_signal_to_process(volname, out, sig):
-    """Sends the signal to one of the process"""
-
-    for line in out.split("\n"):
-        parts = line.split()
-        pid = parts[0]
-        for part in parts:
-            if part.startswith("--volume-id="):
-                if part.split("=")[-1] == volname:
-                    cmd = [ "kill", sig, pid ]
-                    try:
-                        execute(*cmd)
-                    except CommandException as err:
-                        logging.error(logf(
-                            "error to execute command",
-                            volume=volname,
-                            cmd=cmd,
-                            error=format(err)
-                        ))
-                    return
-
-    logging.debug(logf(
-        "Sent SIGHUP to glusterfs process",
-        volname=volname
-    ))
-    return
-
-
 def reload_glusterfs(volume):
```

Comment:
Remove this function itself. Both native and external volfile servers handles notifying the connected clients when any option changes.

Reply:
Removed `reload_glusterfs`.
In csi/main.py. `'reload_glusterfs` is used to reconfigure mounts. Should that be removed too?
L#54 csi/main.py